_health-care/_js/_components/date_input.jsx

```diff
@@ -4,54 +4,72 @@ import _ from 'lodash';
 class DateInput extends React.Component {
   constructor() {
     super();
-    this.state = {month: '', date: '', year: ''};
-    this.handleMonthChange = this.handleMonthChange.bind(this);
-    this.handleDateChange = this.handleDateChange.bind(this);
-    this.handleYearChange = this.handleYearChange.bind(this);
+    this.state = {hasError: false};
+    this.handleChange = this.handleChange.bind(this);
   }
 
   componentWillMount() {
     this.id = _.uniqueId('date-input-');
   }
 
-  handleMonthChange(event) {
-    this.setState({month: event.target.value});
-  }
+  handleChange() {
+    const date = {
+        month: this.refs.month.value,
+        day: this.refs.day.value,
+        year: this.refs.year.value
+      }
+
+    // Update UI based on validation.
+    if (!this.validate(date.day, date.month, date.year)) {
```

Comment:
Wonder if we should also call `validate` on load or something, I noticed that if the page is reloaded and the input field is empty it won't be marked as having an error.